### PR TITLE
AC_PI: set integrator range with [-imax, imax]

### PR DESCRIPTION
**Pull request by xianglunkai**

### CODE REVIEW

**Review by rmackay9 — APPROVED** [review]
This looks correct to me.  The only place that we use the AC_PI integrator is in the IMU heater so hopefully the effect of this bug has been small.

FYI @tridge 

Thanks very much for finding this.

**Review by IamPete1 — CHANGES_REQUESTED** [review]
I agree that conceptually this is correct.

However as @rmackay9 points out the only user is the board heater. Its output must be positive, there is only a heater, it cannot cool. 

The output of the PI controller is used here:
https://github.com/ArduPilot/ardupilot/blob/6fb99d6b1ebc5bf3bbd8edef199fd0bc8b34100f/libraries/AP_BoardConfig/IMU_heater.cpp#L86-L87

The output is constrained in the range 0 to 100. So ever getting a I term outside that range would result in integrator windup against that limit. 

So the 0 is correct. strictly the upper limit should be `min(imax, 100)` too. We default I max to 70 and I doubt anyone has ever changed it. 

The more advanced controllers use external limit flags so the controller does not need to know about the acceptable output range directly. 

The TLDR is that we cannot merge this change. We could add a comment to explain why it is the way it is. I don't think it is worth the more extensive re-work that would be needed to use external limit flags.

**Comment by tridge** [comment]
closing based on comment to @IamPete1
